From 5d02ea32516187256bb9dcf58ad34f2a92c2851b Mon Sep 17 00:00:00 2001
From: xc-racer99 <xc-racer2@live.ca>
Date: Fri, 2 Jan 2015 14:47:25 -0800
Subject: [PATCH 1/2] audio-caf-bfam: Allow devices to not use hwdep
 calibration Some devices do not support this, falcon for sure, so make it
 opt-outable

Inspired by the commit at https://github.com/CyanogenMod/android_hardware_qcom_audio/commit/4f4de4bfb8fcfaec538260a1bed5657d0db1d57a

Change-Id: I7927ecc05fc7177e935f7b7af502a439c0e9524d
---
 hal/Android.mk         |  4 ++++
 hal/msm8974/platform.c | 10 +++++++++-
 2 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/hal/Android.mk b/hal/Android.mk
index 0d8670a..48cc665 100644
--- a/hal/Android.mk
+++ b/hal/Android.mk
@@ -165,6 +165,10 @@ ifeq ($(strip $(AUDIO_FEATURE_PCM_IOCTL_ENABLED)),true)
     LOCAL_CFLAGS += -DPCM_IOCTL_ENABLED
 endif
 
+ifeq ($(strip $(AUDIO_FEATURE_DISABLED_HWDEP_CAL)),true)
+    LOCAL_CFLAGS += -DHWDEP_CAL_DISABLED
+endif
+
 LOCAL_COPY_HEADERS_TO   := mm-audio
 LOCAL_COPY_HEADERS      := audio_extn/audio_defs.h
 
diff --git a/hal/msm8974/platform.c b/hal/msm8974/platform.c
index 93b1775..2df8f38 100644
--- a/hal/msm8974/platform.c
+++ b/hal/msm8974/platform.c
@@ -35,7 +35,9 @@
 #include "voice_extn.h"
 #include "sound/compress_params.h"
 #include "platform_parser.h"
+#ifndef HWDEP_CAL_DISABLED
 #include "sound/msmcal-hwdep.h"
+#endif
 
 #define SOUND_TRIGGER_DEVICE_HANDSET_MONO_LOW_POWER_ACDB_ID (100)
 
@@ -85,11 +87,13 @@
 #define AUDIO_PARAMETER_KEY_VOLUME_BOOST  "volume_boost"
 #define MAX_CAL_NAME 20
 
+#ifndef HWDEP_CAL_DISABLED
 char cal_name_info[WCD9XXX_MAX_CAL][MAX_CAL_NAME] = {
         [WCD9XXX_ANC_CAL] = "anc_cal",
         [WCD9XXX_MBHC_CAL] = "mbhc_cal",
         [WCD9XXX_MAD_CAL] = "mad_cal",
 };
+#endif
 
 enum {
 	VOICE_FEATURE_SET_DEFAULT,
@@ -338,6 +342,8 @@ static int acdb_device_table[SND_DEVICE_MAX] = {
 
 #define DEEP_BUFFER_PLATFORM_DELAY (29*1000LL)
 #define LOW_LATENCY_PLATFORM_DELAY (13*1000LL)
+
+#ifndef HWDEP_CAL_DISABLED
 static int hw_util_open(int card_no)
 {
     int fd = -1;
@@ -422,7 +428,7 @@ static void audio_hwdep_send_cal(struct platform_data *plat_data)
     if (send_codec_cal(acdb_loader_get_calibration, fd) < 0)
         ALOGE("%s: Could not send anc cal", __FUNCTION__);
 }
-
+#endif
 
 
 static void set_echo_reference(struct audio_device *adev, bool enable)
@@ -717,7 +723,9 @@ void *platform_init(struct audio_device *adev)
     /* Read one time ssr property */
     audio_extn_ssr_update_enabled();
     audio_extn_spkr_prot_init(adev);
+#ifndef HWDEP_CAL_DISABLED
     audio_hwdep_send_cal(my_data);
+#endif
     return my_data;
 }
 
-- 
2.1.4


From aa02778010b6a0499af3a6fc5675331bc3ade75f Mon Sep 17 00:00:00 2001
From: Eric Laurent <elaurent@google.com>
Date: Thu, 23 Oct 2014 14:42:59 -0700
Subject: [PATCH 2/2] compress offload: use new sample rate representation.

Pass directly the sample rate value to struct snd_codec
instead of the ALSA enum.

Bug: 17398311.
Change-Id: I79483773807ce3b0b146fde28d6498444c69fe89

Conflicts:
	hal/Android.mk
---
 hal/Android.mk | 4 ++++
 hal/audio_hw.c | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/hal/Android.mk b/hal/Android.mk
index 48cc665..a112577 100644
--- a/hal/Android.mk
+++ b/hal/Android.mk
@@ -110,6 +110,10 @@ ifeq ($(strip $(AUDIO_FEATURE_ENABLED_COMPRESS_CAPTURE)),true)
     LOCAL_SRC_FILES += audio_extn/compress_capture.c
 endif
 
+ifeq ($(strip $(AUDIO_FEATURE_ENABLED_NEW_SAMPLE_RATE)),true)
+    LOCAL_CFLAGS += -DNEW_SAMPLE_RATE_ENABLED
+endif
+
 ifeq ($(strip $(DOLBY_DDP)),true)
     LOCAL_CFLAGS += -DDS1_DOLBY_DDP_ENABLED
     LOCAL_SRC_FILES += audio_extn/dolby.c
diff --git a/hal/audio_hw.c b/hal/audio_hw.c
index 9156a00..be89824 100644
--- a/hal/audio_hw.c
+++ b/hal/audio_hw.c
@@ -2426,8 +2426,12 @@ static int adev_open_output_stream(struct audio_hw_device *dev,
                        platform_get_compress_offload_buffer_size(&config->offload_info);
         }
         out->compr_config.fragments = COMPRESS_OFFLOAD_NUM_FRAGMENTS;
+#ifdef NEW_SAMPLE_RATE_ENABLED
+        out->compr_config.codec->sample_rate = config->offload_info.sample_rate;
+#else
         out->compr_config.codec->sample_rate =
                     compress_get_alsa_rate(config->offload_info.sample_rate);
+#endif
         out->compr_config.codec->bit_rate =
                     config->offload_info.bit_rate;
         out->compr_config.codec->ch_in =
-- 
2.1.4

